### PR TITLE
SHV2: Fix some action tests, some schemas and some logic

### PR DESCRIPTION
**Pull request by Orzelius**
please see individual commits for more details :)

Please note that some changes include logic fixes in addition to tests

### CODE REVIEW

File: core/src/events.ts [hunk 1]
```diff
@@ -223,7 +224,7 @@ export interface Events extends LoggerEvents {
 
     // DEPRECATED: remove in 0.14
     moduleName: string
-    moduleVersion: string
+    moduleVersion: ModuleVersion
```
Comment (vvagaytsev) [inline core/src/events.ts:227]: Do we need to include the whole `ModuleVersion` object in the events? That can be quite heavy. How about storing only `moduleVersionString` here?

File: core/test/unit/src/actions.ts [hunk 2]
```diff
@@ -381,7 +395,7 @@ describe("ActionRouter", () => {
         //   {
         //     name: "test",
         //     dependencies: [],
-        //     disabled: false,
+        //     disabled: false,module actions
```
Comment (vvagaytsev) [inline core/test/unit/src/actions.ts:398]: This looks like some leftover or unfinished comment.

**Comment by vvagaytsev** [comment]
Looks good! 👍  I left a couple of comments, please check.

File: core/src/actions/base.ts [hunk 3]
```diff
@@ -244,7 +244,7 @@ export const baseRuntimeActionConfig = () =>
 
 export const actionStatusSchema = (detailSchema?: Joi.ObjectSchema) =>
   joi.object().keys({
-    status: joi
+    state: joi
```
Comment (vvagaytsev) [inline core/src/actions/base.ts:247]: I think this change is robust, so we can re-generate the docs in the same PR.

Comment (Orzelius) [reply core/src/actions/base.ts:247]: Done in [f92212d](https://github.com/garden-io/garden/pull/3260/commits/f92212dab3ab7385782f726f80f86ad7f53b6e9f)

File: core/src/events.ts (hunk 1 above)
Comment (Orzelius) [reply core/src/events.ts]: makes sense, can always revisit if need be [c6d0d60](https://github.com/garden-io/garden/pull/3260/commits/c6d0d60a8ec97d35bc5c8d47603a60e79dbc3bad)

File: core/test/unit/src/actions.ts (hunk 2 above)
Comment (Orzelius) [reply core/test/unit/src/actions.ts]: oops [706705b](https://github.com/garden-io/garden/pull/3260/commits/706705bfb7a8a0e7ce22863eee94461854364abd)

**Review by vvagaytsev — APPROVED** [review]

**Pull request merged**